pkgs/os-specific/linux/kernel/generic.nix

```
@@ -171,12 +171,23 @@ let
       export HOSTAR=$AR_FOR_BUILD
       export HOSTLD=$LD_FOR_BUILD
 
+      # Absolute paths for tools avoid any PATH-clobbering issues.
+      export CC=${stdenv.cc}/bin/${stdenv.cc.targetPrefix}cc
```

```suggestion
      export CC=${lib.getExe' stdenv.cc "${stdenv.cc.targetPrefix}cc"}
```
is that any better? not sure

---

I would leave it as is to keep the symmetry.

---

I would leave it as is.